Blogpost about Hybrid Search

lizbiella opened the pull request:
Added the blogpost about Hybrid Search

alessandrobenedetti commented:
We are not sure this is the right process to link external blog posts, once we are sure, we'll contribute all Sease's blog posts around Solr (and they are a decent bunch)

janhoy commented on content/pages/blogposts/2024-06-17-hybrid-search.md:
```diff
@@ -0,0 +1,9 @@
+category: solr/blogposts
+slug: hybrid-search-with-apache-solr
+URL: blogposts/hybrid-search-with-apache-solr.html
+save_as: blogposts/hybrid-search-with-apache-solr.html
```
If you add a `summary:` tag, it will be used as into text on the blog home page.

gerlowskija replied:
+1 to Jan's comment.  The `summary` metadata tag will be used as the snippet on the main page (truncated at 512 characters, if it happens to be longer).

janhoy replied:
Please click "Resolve conversation" for all review comments that have been resolved.

janhoy commented on content/pages/blogposts/2024-06-17-hybrid-search.md:
```diff
@@ -0,0 +1,9 @@
+URL: blogposts/hybrid-search-with-apache-solr.html
+save_as: blogposts/hybrid-search-with-apache-solr.html
+
+### This blog shows how to run a hybrid search(keyword-based search + vectors) in Apache Solr with code examples and explanations!
```
The other blog posts use a h1 header, e.g.
```
# Hybrid Search with Apache Solr
```

Then the long-ish text "This blog..." could be plaintext.

gerlowskija replied:
+1 to Jan's formatting tweaks

[0] Less crucially, I think this blurb might advertise the external post a bit better if it had a few more details: who wrote the external post, where is it hosted, a bit more on the "main takeaway", etc.  Maybe consider something like:

> Contrary to popular misconception, Solr already supports hybrid search (the combination of keywork and vector based search techniques)!  Hybrid search can be used in retrieval, ranking, and even Solr's commonly-used "Learning to Rank" module.  Alessandro Benedetti of Sease Ltd. explains more in a recent post, complete with code examples and explanation.
>
> Learn how you can start using hybrid search in your deployments today - the complete post can be read [here](https://sease.io/2023/12/hybrid-search-with-apache-solr.html)

janhoy replied:
Great point. Perhaps we could end up with having a template `.md` file in the repo that folks can start from when they want to make a post that links to external blog.

janhoy reviewed (changes requested):
I spun the site up and found your post did not look like the templates provided. See other comments.
![Skjermbilde 2024-06-20 kl  12 39 56](https://github.com/apache/solr-site/assets/409128/2b440f95-cd7d-49cd-99f2-553a9dc51de1)
![Skjermbilde 2024-06-20 kl  12 40 02](https://github.com/apache/solr-site/assets/409128/1b663adc-53b9-472c-a0ad-fb9bdcda16cb)

I think it will help to add a `summary:` metadata and a level-1 title.

gerlowskija commented:
> We are not sure this is the right process to link external blog posts

Just commenting to affirm - at least IMO, what you guys have done all follows what we landed on in SOLR-16833 👍

gerlowskija reviewed (changes requested):
Marking as 'Request changes' for now, until Jan's formatting comments have been addressed.  But then I'm happy to 'Approve'.

Thanks for adding the first external post to the blog - very exciting!

lizbiella commented:
Thanks @janhoy and @gerlowskija for the comments! I addressed all your concerns and updated the pull request! Let me know if there is something else to change

alessandrobenedetti commented:
Merged! We have plenty (probably >50 Solr blog post in our archive).
Our plan is to contribute one every few day, so the Solr blog will be populated soon :D

Thanks @lizbiella  for the contribution and @janhoy  and @gerlowskija for the reviews, the next pull requests will be easier!

janhoy commented:
Nice @alessandrobenedetti, it is now live on the staging site https://solr.staged.apache.org/blog.html:
![Skjermbilde 2024-06-21 kl  10 46 56](https://github.com/apache/solr-site/assets/409128/a548879d-5558-49c3-8e9c-49a9917d1015)

In order to publish it, you have to merge main branch into production branch. You can do it through a PR or from command line, but make sure it is a normal merge with a merge commit, no squashing or cherry picking.

chatman commented:
> This blog shows how to run a hybrid search (keyword-based search +
vectors) in Apache Solr with code examples and explanations.

Thanks Alessandro. Can we change "blog shows" to "blog post shows" here?

On Fri, 21 Jun, 2024, 2:21 pm Jan Høydahl, ***@***.***> wrote:

> Nice @alessandrobenedetti <https://github.com/alessandrobenedetti>, it is
> now live on the staging site https://solr.staged.apache.org/blog.html:
> Skjermbilde.2024-06-21.kl.10.46.56.png (view on web)
> <https://github.com/apache/solr-site/assets/409128/a548879d-5558-49c3-8e9c-49a9917d1015>
>
> In order to publish it, you have to merge main branch into production
> branch. You can do it through a PR or from command line, but make sure it
> is a normal merge with a merge commit, no squashing or cherry picking.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/solr-site/pull/100#issuecomment-2182309171>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDCR5BFJWUL7VH6E5U55J3ZIPSP5AVCNFSM6AAAAABJOB5UMKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOBSGMYDSMJXGE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

alessandrobenedetti commented:
> This blog shows how to run a hybrid search (keyword-based search +
> vectors) in Apache Solr with code examples and explanations. Thanks Alessandro. Can we change "blog shows" to "blog post shows" here?
> […](#)
> On Fri, 21 Jun, 2024, 2:21 pm Jan Høydahl, ***@***.***> wrote: Nice @alessandrobenedetti <https://github.com/alessandrobenedetti>, it is now live on the staging site https://solr.staged.apache.org/blog.html: Skjermbilde.2024-06-21.kl.10.46.56.png (view on web) <https://github.com/apache/solr-site/assets/409128/a548879d-5558-49c3-8e9c-49a9917d1015> In order to publish it, you have to merge main branch into production branch. You can do it through a PR or from command line, but make sure it is a normal merge with a merge commit, no squashing or cherry picking. — Reply to this email directly, view it on GitHub <[#100 (comment)](https://github.com/apache/solr-site/pull/100#issuecomment-2182309171)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABDCR5BFJWUL7VH6E5U55J3ZIPSP5AVCNFSM6AAAAABJOB5UMKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOBSGMYDSMJXGE> . You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

thanks @chatman , just merged this fix into the production branch!
